File: 007.clj
; What is the 10 001st prime number?

(defn factor? [n d]
    (zero? (rem n d)))

(defn factors [n]
    (filter #(factor? n %) (range 2 n)))

(defn prime? [n]
    (empty? (factors n)))

(nth (filter prime? (range)) 10002)

File: 007.hs
-- What is the 10,001st prime number?

module Main where

factor n d
  | mod n d == 0 = True
  | otherwise = False

factors n = [d | d <- [2..n-1], mod n d == 0]

prime n = null (factors n)

main = print $ filter prime [1..] !! 10001

-- 104743
